### ISSUE TITLE
Connection lost & workspace needs to be reloaded when debugging theme / extension after a certain time

### ISSUE THREAD

**Issue by Foxhoundn** [gitpod-io/gitpod#7313]
### Bug description

I have 2 workspaces in which I work on either a theme or a VS Code extension - both of these and only these workspaces are randomly disconnected after a certain random time - sometimes it's a few minutes, sometimes it's 20 minutes. I was finally able to get some log data of the console before the window was reloaded.

### Steps to reproduce

You could reproduce when working on a theme or any VS Code extension I assume - that's where I reproduced.

You can fork & test with my theme - https://github.com/Foxhoundn/synthax-code-theme
Or you can try forking our extension, but bear in mind this is a huge project and will require additional steps to set up - https://github.com/qoretechnologies/qorus-vscode

After a certain time working with the other window open (when Run & Debug is running), the workspace disconnects - here are some of the logs

```
 WARN Error: Aborted onWillSaveTextDocument-event after 1750ms
    at mainThreadSaveParticipant.ts:39
mainThreadExtensionService.ts:63 [[object Object]]Cannot find target e2c95a853c53a07a40ef9bbf
$onExtensionRuntimeError @ mainThreadExtensionService.ts:63
mainThreadExtensionService.ts:64 Error: Cannot find target e2c95a853c53a07a40ef9bbf
	at t.SessionManager.createNewChildSession (/ide/extensions/ms-vscode.js-debug/src/extension.js:2)
	at /ide/extensions/ms-vscode.js-debug/src/extension.js:2
	at Server.<anonymous> (/ide/extensions/ms-vscode.js-debug/src/extension.js:2)
	at Server.emit (/events.js:315)
	at TCP.onconnection (/net.js:1560)
$onExtensionRuntimeError @ mainThreadExtensionService.ts:64
mainThreadExtensionService.ts:63 [[object Object]]Cannot find target e2c95a853c53a07a40ef9bbf
$onExtensionRuntimeError @ mainThreadExtensionService.ts:63
mainThreadExtensionService.ts:64 Error: Cannot find target e2c95a853c53a07a40ef9bbf
	at t.SessionManager.createNewChildSession (/ide/extensions/ms-vscode.js-debug/src/extension.js:2)
	at /ide/extensions/ms-vscode.js-debug/src/extension.js:2
	at Server.<anonymous> (/ide/extensions/ms-vscode.js-debug/src/extension.js:2)
	at Server.emit (/events.js:315)
	at TCP.onconnection (/net.js:1560)
$onExtensionRuntimeError @ mainThreadExtensionService.ts:64
console.ts:137 [Extension Host] rejected promise not handled within 1 second: TypeError: Cannot read property 'text' of undefined
I @ console.ts:137
console.ts:137 [Extension Host] stack trace: TypeError: Cannot read property 'text' of undefined	at /home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17:66873	at Object.t.default (/home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17:67000)	at t.default (/home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17:75159)	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at P.$acceptDirtyStateChanged (/ide/out/vs/server/remoteExtensionHostProcess.js:95:71823)	at t._doInvokeHandler (/ide/out/vs/server/remoteExtensionHostProcess.js:99:13802)	at t._invokeHandler (/ide/out/vs/server/remoteExtensionHostProcess.js:99:13486)	at t._receiveRequest (/ide/out/vs/server/remoteExtensionHostProcess.js:99:12148)	at t._receiveOneMessage (/ide/out/vs/server/remoteExtensionHostProcess.js:99:10826)	at /ide/out/vs/server/remoteExtensionHostProcess.js:99:8922	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at r.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:74:18942)	at /ide/out/vs/server/remoteExtensionHostProcess.js:114:34341	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at r.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:74:18942)	at n._receiveMessage (/ide/out/vs/server/remoteExtensionHostProcess.js:74:23523)	at /ide/out/vs/server/remoteExtensionHostProcess.js:74:21057	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at w.acceptChunk (/ide/out/vs/server/remoteExtensionHostProcess.js:74:15770)	at /ide/out/vs/server/remoteExtensionHostProcess.js:74:14900	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at s._processReadQueue (/ide/out/vs/server/remoteExtensionHostProcess.js:114:21295)
I @ console.ts:137
mainThreadExtensionService.ts:63 [[object Object]]Cannot read property 'text' of undefined
$onExtensionRuntimeError @ mainThreadExtensionService.ts:63
mainThreadExtensionService.ts:64 TypeError: Cannot read property 'text' of undefined
	at /home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17
	at Object.t.default (/home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17)
	at t.default (/home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17)
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at P.$acceptDirtyStateChanged (/ide/out/vs/server/remoteExtensionHostProcess.js:95)
	at t._doInvokeHandler (/ide/out/vs/server/remoteExtensionHostProcess.js:99)
	at t._invokeHandler (/ide/out/vs/server/remoteExtensionHostProcess.js:99)
	at t._receiveRequest (/ide/out/vs/server/remoteExtensionHostProcess.js:99)
	at t._receiveOneMessage (/ide/out/vs/server/remoteExtensionHostProcess.js:99)
	at /ide/out/vs/server/remoteExtensionHostProcess.js:99
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at r.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:74)
	at /ide/out/vs/server/remoteExtensionHostProcess.js:114
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at r.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:74)
	at n._receiveMessage (/ide/out/vs/server/remoteExtensionHostProcess.js:74)
	at /ide/out/vs/server/remoteExtensionHostProcess.js:74
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at w.acceptChunk (/ide/out/vs/server/remoteExtensionHostProcess.js:74)
	at /ide/out/vs/server/remoteExtensionHostProcess.js:74
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at s._processReadQueue (/ide/out/vs/server/remoteExtensionHostProcess.js:114)
$onExtensionRuntimeError @ mainThreadExtensionService.ts:64
console.ts:137 [Extension Host] rejected promise not handled within 1 second: TypeError: Cannot read property 'text' of undefined
I @ console.ts:137
console.ts:137 [Extension Host] stack trace: TypeError: Cannot read property 'text' of undefined	at /home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17:66873	at Object.t.default (/home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17:67000)	at t.default (/home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17:75159)	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at P.$acceptDirtyStateChanged (/ide/out/vs/server/remoteExtensionHostProcess.js:95:71823)	at P.$acceptModelSaved (/ide/out/vs/server/remoteExtensionHostProcess.js:95:71566)	at t._doInvokeHandler (/ide/out/vs/server/remoteExtensionHostProcess.js:99:13802)	at t._invokeHandler (/ide/out/vs/server/remoteExtensionHostProcess.js:99:13486)	at t._receiveRequest (/ide/out/vs/server/remoteExtensionHostProcess.js:99:12148)	at t._receiveOneMessage (/ide/out/vs/server/remoteExtensionHostProcess.js:99:10826)	at /ide/out/vs/server/remoteExtensionHostProcess.js:99:8922	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at r.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:74:18942)	at /ide/out/vs/server/remoteExtensionHostProcess.js:114:34341	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at r.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:74:18942)	at n._receiveMessage (/ide/out/vs/server/remoteExtensionHostProcess.js:74:23523)	at /ide/out/vs/server/remoteExtensionHostProcess.js:74:21057	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at w.acceptChunk (/ide/out/vs/server/remoteExtensionHostProcess.js:74:15770)	at /ide/out/vs/server/remoteExtensionHostProcess.js:74:14900	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66:1712)	at s._processReadQueue (/ide/out/vs/server/remoteExtensionHostProcess.js:114:21295)
I @ console.ts:137
mainThreadExtensionService.ts:63 [[object Object]]Cannot read property 'text' of undefined
$onExtensionRuntimeError @ mainThreadExtensionService.ts:63
mainThreadExtensionService.ts:64 TypeError: Cannot read property 'text' of undefined
	at /home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17
	at Object.t.default (/home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17)
	at t.default (/home/gitpod/.vscode-remote/extensions/tabnine.tabnine-vscode-3.5.11/out/extension.js:17)
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at P.$acceptDirtyStateChanged (/ide/out/vs/server/remoteExtensionHostProcess.js:95)
	at P.$acceptModelSaved (/ide/out/vs/server/remoteExtensionHostProcess.js:95)
	at t._doInvokeHandler (/ide/out/vs/server/remoteExtensionHostProcess.js:99)
	at t._invokeHandler (/ide/out/vs/server/remoteExtensionHostProcess.js:99)
	at t._receiveRequest (/ide/out/vs/server/remoteExtensionHostProcess.js:99)
	at t._receiveOneMessage (/ide/out/vs/server/remoteExtensionHostProcess.js:99)
	at /ide/out/vs/server/remoteExtensionHostProcess.js:99
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at r.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:74)
	at /ide/out/vs/server/remoteExtensionHostProcess.js:114
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at r.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:74)
	at n._receiveMessage (/ide/out/vs/server/remoteExtensionHostProcess.js:74)
	at /ide/out/vs/server/remoteExtensionHostProcess.js:74
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at w.acceptChunk (/ide/out/vs/server/remoteExtensionHostProcess.js:74)
	at /ide/out/vs/server/remoteExtensionHostProcess.js:74
	at u.fire (/ide/out/vs/server/remoteExtensionHostProcess.js:66)
	at s._processReadQueue (/ide/out/vs/server/remoteExtensionHostProcess.js:114)
$onExtensionRuntimeError @ mainThreadExtensionService.ts:64
mainThreadExtensionService.ts:63 [[object Object]]Cannot find target e2c95a853c53a07a40ef9bbf
$onExtensionRuntimeError @ mainThreadExtensionService.ts:63
mainThreadExtensionService.ts:64 Error: Cannot find target e2c95a853c53a07a40ef9bbf
	at t.SessionManager.createNewChildSession (/ide/extensions/ms-vscode.js-debug/src/extension.js:2)
	at /ide/extensions/ms-vscode.js-debug/src/extension.js:2
	at Server.<anonymous> (/ide/extensions/ms-vscode.js-debug/src/extension.js:2)
	at Server.emit (/events.js:315)
	at TCP.onconnection (/net.js:1560)
$onExtensionRuntimeError @ mainThreadExtensionService.ts:64
5errors.ts:160 Uncaught (in promise) Canceled: Canceled
    at y (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:8:1157)
    at https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:1736:2563
    at l.fire (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:1712)
    at M.cancel (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:15007)
    at S.cancel (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:15548)
    at S.dispose (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:15603)
    at p (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:578:33730)
    at https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:578:34817
    at l.fire (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:1712)
    at https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:555:31586
y @ errors.ts:160
(anonymous) @ searchService.ts:163
fire @ event.ts:577
cancel @ cancellation.ts:69
cancel @ cancellation.ts:124
dispose @ cancellation.ts:130
p @ pickerQuickAccess.ts:120
(anonymous) @ pickerQuickAccess.ts:269
fire @ event.ts:577
(anonymous) @ quickInput.ts:745
fire @ event.ts:577
onValueChange @ inputBox.ts:527
(anonymous) @ inputBox.ts:219
await in (anonymous) (async)
(anonymous) @ pickerQuickAccess.ts:269
fire @ event.ts:577
(anonymous) @ quickInput.ts:745
fire @ event.ts:577
onValueChange @ inputBox.ts:527
(anonymous) @ inputBox.ts:219
2errors.ts:160 Uncaught (in promise) Canceled: Canceled
    at y (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:8:1157)
    at https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:1736:2563
    at l.fire (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:1712)
    at M.cancel (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:15007)
    at S.cancel (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:15548)
    at l.fire (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:1732)
    at M.cancel (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:15007)
    at S.cancel (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:15548)
    at https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:584:32698
    at l.fire (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:1712)
y @ errors.ts:160
(anonymous) @ searchService.ts:163
fire @ event.ts:577
cancel @ cancellation.ts:69
cancel @ cancellation.ts:124
fire @ event.ts:579
cancel @ cancellation.ts:69
cancel @ cancellation.ts:124
(anonymous) @ quickAccess.ts:139
fire @ event.ts:577
didHide @ quickInput.ts:311
hide @ quickInput.ts:1697
(anonymous) @ quickInput.ts:1309
fire @ event.ts:577
(anonymous) @ dom.ts:986
await in (anonymous) (async)
(anonymous) @ pickerQuickAccess.ts:269
fire @ event.ts:577
(anonymous) @ quickInput.ts:745
fire @ event.ts:577
onValueChange @ inputBox.ts:527
(anonymous) @ inputBox.ts:219
log.ts:301  INFO [remote-connection][ExtensionHost][bdb08…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
log.ts:313   ERR CloseEvent
log.ts:301  INFO [remote-connection][ExtensionHost][bdb08…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
log.ts:301  INFO [remote-connection][ExtensionHost][bdb08…][reconnect] resolving connection...
log.ts:301  INFO [remote-connection][ExtensionHost][bdb08…][reconnect] connecting to black-marmoset-d720tsvs.ws-eu23.gitpod.io:443...
log.ts:313   ERR [remote-connection][ExtensionHost][bdb08…][reconnect][black-marmoset-d720tsvs.ws-eu23.gitpod.io:443] received error control message when negotiating connection. Error:
log.ts:313   ERR Error: Connection error: Unknown reconnection token (seen before)
    at W (remoteAgentConnection.ts:743)
    at remoteAgentConnection.ts:325
    at l.fire (event.ts:577)
    at o.fire (ipc.net.ts:638)
    at u._receiveMessage (ipc.net.ts:964)
    at ipc.net.ts:899
    at l.fire (event.ts:577)
    at r.acceptChunk (ipc.net.ts:382)
    at ipc.net.ts:338
    at browserSocketFactory.ts:230
log.ts:313   ERR [remote-connection][ExtensionHost][bdb08…][reconnect] A permanent error occurred in the reconnecting loop! Will give up now! Error:
log.ts:313   ERR Error: Connection error: Unknown reconnection token (seen before)
    at W (remoteAgentConnection.ts:743)
    at remoteAgentConnection.ts:325
    at l.fire (event.ts:577)
    at o.fire (ipc.net.ts:638)
    at u._receiveMessage (ipc.net.ts:964)
    at ipc.net.ts:899
    at l.fire (event.ts:577)
    at r.acceptChunk (ipc.net.ts:382)
    at ipc.net.ts:338
    at browserSocketFactory.ts:230
abstractExtensionService.ts:644 Extension host (Remote) terminated unexpectedly. Code: 0, Signal: null
_onExtensionHostCrashed @ abstractExtensionService.ts:644
await in _onExtensionHostCrashed (async)
_beginReconnecting @ remoteAgentConnection.ts:594
(anonymous) @ remoteAgentConnection.ts:572
fire @ event.ts:577
fire @ ipc.net.ts:638
(anonymous) @ ipc.net.ts:832
n @ browserSocketFactory.ts:238
fire @ event.ts:577
(anonymous) @ browserSocketFactory.ts:174
e._callEventListener @ reconnecting-websocket-mjs.js:555
(anonymous) @ reconnecting-websocket-mjs.js:195
_handleClose @ reconnecting-websocket-mjs.js:195
5errors.ts:160 Uncaught (in promise) Canceled: Canceled
    at Object.y [as canceled] (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:8:1157)
    at h._remoteCall (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:1535:13193)
    at Proxy.x.<computed>.A.charCodeAt.x.<computed> (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:1535:9588)
    at https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:1528:2495
    at l.fire (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:69:1712)
    at https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:571:43727
    at https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:17:30315
    at Object.dispose (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:9:7058)
    at n.dispose (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:1528:2647)
    at Xr.dispose (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:1586:6178)
y @ errors.ts:160
_remoteCall @ rpcProtocol.ts:477
x.<computed>.A.charCodeAt.x.<computed> @ rpcProtocol.ts:252
(anonymous) @ mainThreadLanguageFeatures.ts:56
fire @ event.ts:577
(anonymous) @ languageConfigurationRegistry.ts:198
(anonymous) @ lifecycle.ts:170
(anonymous) @ functional.ts:17
dispose @ mainThreadLanguageFeatures.ts:68
dispose @ extensionHostManager.ts:182
_onExtensionHostCrashed @ abstractExtensionService.ts:650
_onExtensionHostCrashOrExit @ abstractExtensionService.ts:636
(anonymous) @ abstractExtensionService.ts:626
fire @ event.ts:577
_onExtHostConnectionLost @ remoteExtensionHost.ts:209
(anonymous) @ remoteExtensionHost.ts:137
fire @ event.ts:577
fire @ ipc.net.ts:638
acceptDisconnect @ ipc.net.ts:922
F @ remoteAgentConnection.ts:732
_gotoPermanentFailure @ remoteAgentConnection.ts:687
(anonymous) @ remoteAgentConnection.ts:522
triggerPermanentFailure @ remoteAgentConnection.ts:522
_runReconnectingLoop @ remoteAgentConnection.ts:643
await in _runReconnectingLoop (async)
_beginReconnecting @ remoteAgentConnection.ts:594
(anonymous) @ remoteAgentConnection.ts:572
fire @ event.ts:577
fire @ ipc.net.ts:638
(anonymous) @ ipc.net.ts:832
n @ browserSocketFactory.ts:238
fire @ event.ts:577
(anonymous) @ browserSocketFactory.ts:174
e._callEventListener @ reconnecting-websocket-mjs.js:555
(anonymous) @ reconnecting-websocket-mjs.js:195
_handleClose @ reconnecting-websocket-mjs.js:195
search.ts:460 Uncaught (in promise) Error: Canceled
    at L (search.ts:468)
    at searchService.ts:237
    at async searchService.ts:142
    at async cacheState.ts:71
D @ search.ts:460
L @ search.ts:468
(anonymous) @ searchService.ts:237
await in (anonymous) (async)
load @ cacheState.ts:82
createFileQueryCache @ anythingQuickAccess.ts:495
set @ anythingQuickAccess.ts:117
provide @ anythingQuickAccess.ts:210
doShowOrPick @ quickAccess.ts:132
show @ quickAccess.ts:40
handler @ quickAccessActions.ts:138
w.handler @ commands.ts:90
invokeFunction @ instantiationService.ts:61
_tryExecuteCommand @ commandService.ts:84
(anonymous) @ commandService.ts:73
Promise.then (async)
executeCommand @ commandService.ts:73
_doDispatch @ abstractKeybindingService.ts:296
_dispatch @ abstractKeybindingService.ts:187
(anonymous) @ keybindingService.ts:261
DevTools failed to load source map: Could not load content for https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/node_modules/vscode-textmate/release/main.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load source map: Could not load content for https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/node_modules/xterm/lib/xterm.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load source map: Could not load content for https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/node_modules/xterm-addon-unicode11/lib/xterm-addon-unicode11.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load source map: Could not load content for https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/node_modules/xterm-addon-search/lib/xterm-addon-search.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load source map: Could not load content for https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/node_modules/@gitpod/local-app-api-grpcweb/lib/localapp.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
log.ts:313   ERR Canceled: Canceled
    at Object.y [as canceled] (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:8:1157)
    at Object.call (https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:85:1927)
    at https://blobserve.ws-eu23.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-0d1e8cc5dcc8ccb38efcc2cd6404316b6e7bf4e9/__files__/out/vs/workbench/workbench.web.api.js:85:7223
```

I also have screenshots if for whatever reason you'd need those.

### Workspace affected

black-marmoset-d720tsvs

### Expected behavior

Connection should not be terminated.

### Example repository

https://github.com/qoretechnologies/qorus-vscode

### Anything else?

Only happens when Run & Debug is running (theme or extension) and the 2nd code tab is opened.

**Comment by jeanp413** [comment]
I could reproduce this but it took a almost an hour =S
```
 INFO [remote-connection][ExtensionHost][981ee…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
workbench.web.api.js:577   ERR CloseEvent {isTrusted: true, wasClean: false, code: 1006, reason: '', type: 'close', …}isTrusted: truebubbles: falsecancelBubble: falsecancelable: falsecode: 1006composed: falsecurrentTarget: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…e8c6&reconnection=false&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}defaultPrevented: falseeventPhase: 0path: []reason: ""returnValue: truesrcElement: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…e8c6&reconnection=false&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}target: WebSocketbinaryType: "blob"bufferedAmount: 0extensions: "permessage-deflate"onclose: nullonerror: nullonmessage: nullonopen: nullprotocol: ""readyState: 3url: "wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?reconnectionToken=981eeb9d-7754-4c98-9026-2b0afdf5e8c6&reconnection=false&skipWebSocketFrames=false"[[Prototype]]: WebSockettimeStamp: 3880320.199999988type: "close"wasClean: false[[Prototype]]: CloseEvent
workbench.web.api.js:577  INFO [remote-connection][ExtensionHost][981ee…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
workbench.web.api.js:577  INFO [remote-connection][ExtensionHost][981ee…][reconnect] resolving connection...
workbench.web.api.js:577  INFO [remote-connection][ExtensionHost][981ee…][reconnect] connecting to red-donkey-1sl9sf6h.ws-us25.gitpod.io:443...
workbench.web.api.js:577  INFO [remote-connection][ExtensionHost][981ee…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
workbench.web.api.js:577   ERR CloseEvent {isTrusted: true, wasClean: false, code: 1006, reason: '', type: 'close', …}isTrusted: truebubbles: falsecancelBubble: falsecancelable: falsecode: 1006composed: falsecurrentTarget: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…5e8c6&reconnection=true&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}defaultPrevented: falseeventPhase: 0path: []reason: ""returnValue: truesrcElement: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…5e8c6&reconnection=true&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}target: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…5e8c6&reconnection=true&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}timeStamp: 3882028.399999976type: "close"wasClean: false[[Prototype]]: CloseEvent
workbench.web.api.js:577   ERR [remote-connection][ExtensionHost][981ee…][reconnect][red-donkey-1sl9sf6h.ws-us25.gitpod.io:443] received error control message when negotiating connection. Error:
workbench.web.api.js:577   ERR Error: Connection error: Unknown reconnection token (seen before)
    at W (workbench.web.api.js:584)
    at workbench.web.api.js:584
    at l.fire (workbench.web.api.js:69)
    at r.fire (workbench.web.api.js:85)
    at d._receiveMessage (workbench.web.api.js:85)
    at workbench.web.api.js:85
    at l.fire (workbench.web.api.js:69)
    at s.acceptChunk (workbench.web.api.js:85)
    at workbench.web.api.js:85
    at workbench.web.api.js:584
workbench.web.api.js:577   ERR [remote-connection][ExtensionHost][981ee…][reconnect] A permanent error occurred in the reconnecting loop! Will give up now! Error:
workbench.web.api.js:577   ERR Error: Connection error: Unknown reconnection token (seen before)
    at W (workbench.web.api.js:584)
    at workbench.web.api.js:584
    at l.fire (workbench.web.api.js:69)
    at r.fire (workbench.web.api.js:85)
    at d._receiveMessage (workbench.web.api.js:85)
    at workbench.web.api.js:85
    at l.fire (workbench.web.api.js:69)
    at s.acceptChunk (workbench.web.api.js:85)
    at workbench.web.api.js:85
    at workbench.web.api.js:584
```

**Comment by Foxhoundn** [comment]
> I could reproduce this but it took almost an hour =S, need to investigate more...
> 
> ```
>  INFO [remote-connection][ExtensionHost][981ee…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
> workbench.web.api.js:577   ERR CloseEvent {isTrusted: true, wasClean: false, code: 1006, reason: '', type: 'close', …}isTrusted: truebubbles: falsecancelBubble: falsecancelable: falsecode: 1006composed: falsecurrentTarget: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…e8c6&reconnection=false&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}defaultPrevented: falseeventPhase: 0path: []reason: ""returnValue: truesrcElement: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…e8c6&reconnection=false&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}target: WebSocketbinaryType: "blob"bufferedAmount: 0extensions: "permessage-deflate"onclose: nullonerror: nullonmessage: nullonopen: nullprotocol: ""readyState: 3url: "wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?reconnectionToken=981eeb9d-7754-4c98-9026-2b0afdf5e8c6&reconnection=false&skipWebSocketFrames=false"[[Prototype]]: WebSockettimeStamp: 3880320.199999988type: "close"wasClean: false[[Prototype]]: CloseEvent
> workbench.web.api.js:577  INFO [remote-connection][ExtensionHost][981ee…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
> workbench.web.api.js:577  INFO [remote-connection][ExtensionHost][981ee…][reconnect] resolving connection...
> workbench.web.api.js:577  INFO [remote-connection][ExtensionHost][981ee…][reconnect] connecting to red-donkey-1sl9sf6h.ws-us25.gitpod.io:443...
> workbench.web.api.js:577  INFO [remote-connection][ExtensionHost][981ee…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
> workbench.web.api.js:577   ERR CloseEvent {isTrusted: true, wasClean: false, code: 1006, reason: '', type: 'close', …}isTrusted: truebubbles: falsecancelBubble: falsecancelable: falsecode: 1006composed: falsecurrentTarget: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…5e8c6&reconnection=true&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}defaultPrevented: falseeventPhase: 0path: []reason: ""returnValue: truesrcElement: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…5e8c6&reconnection=true&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}target: WebSocket {url: 'wss://red-donkey-1sl9sf6h.ws-us25.gitpod.io/?recon…5e8c6&reconnection=true&skipWebSocketFrames=false', readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}timeStamp: 3882028.399999976type: "close"wasClean: false[[Prototype]]: CloseEvent
> workbench.web.api.js:577   ERR [remote-connection][ExtensionHost][981ee…][reconnect][red-donkey-1sl9sf6h.ws-us25.gitpod.io:443] received error control message when negotiating connection. Error:
> workbench.web.api.js:577   ERR Error: Connection error: Unknown reconnection token (seen before)
>     at W (workbench.web.api.js:584)
>     at workbench.web.api.js:584
>     at l.fire (workbench.web.api.js:69)
>     at r.fire (workbench.web.api.js:85)
>     at d._receiveMessage (workbench.web.api.js:85)
>     at workbench.web.api.js:85
>     at l.fire (workbench.web.api.js:69)
>     at s.acceptChunk (workbench.web.api.js:85)
>     at workbench.web.api.js:85
>     at workbench.web.api.js:584
> workbench.web.api.js:577   ERR [remote-connection][ExtensionHost][981ee…][reconnect] A permanent error occurred in the reconnecting loop! Will give up now! Error:
> workbench.web.api.js:577   ERR Error: Connection error: Unknown reconnection token (seen before)
>     at W (workbench.web.api.js:584)
>     at workbench.web.api.js:584
>     at l.fire (workbench.web.api.js:69)
>     at r.fire (workbench.web.api.js:85)
>     at d._receiveMessage (workbench.web.api.js:85)
>     at workbench.web.api.js:85
>     at l.fire (workbench.web.api.js:69)
>     at s.acceptChunk (workbench.web.api.js:85)
>     at workbench.web.api.js:85
>     at workbench.web.api.js:584
> ```

Amazing! I am glad It's not just me, thank you :)

Yesterday it didn't happen for 2 hours, then it started happening every ~30 mins.

**Comment by jeanp413** [comment]
more logs from the server
```
[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr> 
<--- Last few GCs --->

[7665:0x4d811c0]  3527326 ms: Mark-sweep (reduce) 2044.8 (2051.0) -> 2043.9 (2052.3) MB, 2110.7 / 0.0 ms  (average mu = 0.094, current mu = 0.003) allocation failure scavenge might not succeed
[7665:0x4d811c0]  3529563 ms: Mark-sweep (reduce) 2045.0 (2051.3) -> 2044.2 (2052.3) MB, 2232.5 / 0.0 ms  (average mu = 0.049, current mu = 0.002) allocation failure scavenge might not succeed


<--- JS stacktrace --->



[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr> FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr>  1: 0xa04200 node::Abort() [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr>  2: 0x94e4e9 node::FatalError(char const*, char const*) [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr>  3: 0xb7978e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr>  4: 0xb79b07 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr>  5: 0xd34395  [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr>  6: 0xd34f1f  [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr>  7: 0xd42fab v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr>  8: 0xd46b6c v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr>  9: 0xd1524b v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr> 10: 0x105b23f v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:49] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665><stderr> 11: 0x1401219  [/workspace/vscode/.build/node/v14.16.0/linux-x64/node]


[14:47:50] [127.0.0.1][a7a40b3a][ExtensionHostConnection] <8429><stderr> Debugger listening on ws://0.0.0.0:53683/d0a650e1-55d5-4856-b031-fde982fd3dbf
For help, see: https://nodejs.org/en/docs/inspector


[14:47:50] [127.0.0.1][6c91d826][ExtensionHostConnection] <7665> Extension Host Process exited with code: null, signal: SIGABRT.
```

**Comment by jeanp413** [comment]
Reopening as this is fixed but not deployed to production yet.
@loujaybee how do we keep track if changes are deployed or not in the new project view?

**Comment by jeanp413** [comment]
This is deployed now :tada:

**Comment by Foxhoundn** [comment]
@jeanp413 this has started happening again (now using VSCode Desktop) - the debugged extension just randomly shuts down and `Establishing SSH connection` notification appears in the main VSCode window.

Also this error appears in the browser where gitpod was launched from

```
Access to internal resource at 'https://blobserve.ws-eu42.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-e4d4c1cc703e3adc62e4cfe935f37d5d2eeb6df3/__files__/manifest.json' (redirected from 'https://qoretechnologie-qorusvsc-t90myr0jca2.ws-eu42.gitpod.io/manifest.json') from origin 'https://qoretechnologie-qorusvsc-t90myr0jca2.ws-eu42.gitpod.io' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'.

10:00:06.148 DevTools failed to load source map: Could not load content for https://blobserve.ws-eu42.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-e4d4c1cc703e3adc62e4cfe935f37d5d2eeb6df3/__files__/node_modules/@gitpod/local-app-api-grpcweb/lib/localapp.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

**Comment by jeanp413** [comment]
Hi @Foxhoundn could you create a new issue for this :pray: ? 
this is different as vscode desktop uses remote ssh extension, also could you get some logs in the output windows for Gitpod and remote server
Thanks
